File: bubble_sort.py
import random

def bubble_sort(arr, descending=False):
    n = len(arr)
    for i in range(n):
        for j in range(0, n - i - 1):
            if (not descending and arr[j] > arr[j + 1]) or (descending and arr[j] < arr[j + 1]):
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
    return arr


if __name__ == "__main__":
    nums = [random.randint(1, 100) for _ in range(10)]
    print("Original array:", nums)

    sorted_asc = bubble_sort(nums.copy(), descending=False)
    print("Sorted ascending:", sorted_asc)

    sorted_desc = bubble_sort(nums.copy(), descending=True)
    print("Sorted descending:", sorted_desc)

File: sortTechniques/BubbleSort.java
package sortTechniques;

public class BubbleSort {
    public static void main(String[] args) {
        int[] arr = {12,34,56,7,43};
        for(int i=0;i<arr.length;i++){
            for(int j=i;j<arr.length-1;j++){
                if(arr[i]>arr[i+1]){
                int temp;
                temp = arr[i];
                arr[i] = arr[i+1];
                arr[i+1] = temp;
            }}
        }
        for (int i : arr) {
            System.out.println(i); 
         }
    }
}
